internal/handler/userHandler.go

```diff
@@ -82,24 +94,34 @@ func (h *Handler) Save(context *gin.Context) {
 // @Description Find user by provided id
 // @Tags Users
 // @Produce json
-// @Param id path int true "User id"
-// @Success 201 {object} response.UserResponseDto "User Found successfully"
-// @Failure 422 {object} response.ErrorDto "Id in the wrong format"
+// @Param id path string true "User id"
+// @Failure 422 {object} response.ErrorDto "ID in the wrong format"
 // @Failure 404 {object} response.ErrorDto "User not found"
+// @Failure 500 {object} response.ErrorDto "Failed to connect to database"
 // @Router /users/{id} [get]
 func (h *Handler) FindById(context *gin.Context) {
 	id := context.Param("id")
 	logger.Infof("Finding user by id %s", id)
 	user, err := h.service.GetUserById(context, id)
-	if err != nil {
+	if user == nil {
+		var databaseError *exceptions.DatabaseError
+		if errors.As(err, &databaseError) {
```

Comment:
We should check if the error is nil, not if it's a specific type. If it's not nil we know we didn't find the user reguardless of the error type.

Reply:
But if I check err == nil, when success the error will be nil and will pass the if clause throwing an exception. So I check first for user nil and after if the error is not nil and what type of error it is.

Reply:
In order to avoid nested if statements you should something like the following.
```user, err := h.service.GetUserById(context, id)
if err != nil {
	// handle the error case
}
if user == nil {
	// handle user not found because it was not in the database case
}
// Return the found user```